application/helpers/expressions/em_core_helper.php

```diff
@@ -325,7 +325,7 @@ public function RDP_EvaluateBinary(array $token)
         // Not sure if needed to test if [2] is set. : TODO review
         if ($bBothNumeric) {
             $aForceStringArray = array('DQ_STRING', 'DS_STRING', 'STRING'); // Question can return NUMBER or WORD : DQ and DS is string entered by user, STRING is a result of a String function
-            if ((isset($arg1[2]) && in_array($arg1[2], $aForceStringArray) || (isset($arg2[2]) && in_array($arg2[2], $aForceStringArray)))) {
+            if ((isset($arg1[2]) && in_array($arg1[2], $aForceStringArray) && (isset($arg2[2]) && in_array($arg2[2], $aForceStringArray)))) {
```

Comment:
For me, there's absolutely no way to know if this change breaks anything without a complete test suite. :|

Reply:
The issue is (in 3.19) : JS and PHP does 2 different think.

See the 1st commit : https://github.com/LimeSurvey/LimeSurvey/pull/1335/commits/84e7ebe993b6f879795085a651689387152e0217
I test ONLY if js value == php value